File: src/models/assignment-return.ts
import type { Attachment } from "./attachment";
import type { AssignmentReturnKind } from "./assignment-return-kind";

export type AssignmentReturn = {
  readonly kind: AssignmentReturnKind
  /**
   * File that the user uploaded.
   */
  uploaded?: Attachment
  /**
   * Whether the user can upload a file or not.
   */
  canUpload: boolean
  // NOTE: Not sure if we can block the upload once the user has uploaded a file (so the property updates from `true` to `false`)
  // If we cannot, then this should be readonly.
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/AssignmentReturn.kt
package ink.literate.pawnote.models

data class AssignmentReturn(
    val kind: AssignmentReturnKind,
    /**
     * File that the user uploaded.
     */
    var uploaded: Attachment? = null,
    /**
     * Whether the user can upload a file or not.
     */
    var canUpload: Boolean
    // NOTE: Not sure if we can block the upload once the user has uploaded a file (so the property updates from `true` to `false`)
    // If we cannot, then this should be readonly.
)